src/response.rs

```diff
@@ -16,13 +17,20 @@ use http_client;
 
 /// An HTTP response, returned by `Request`.
 pub struct Response {
+    cookies: Vec<Cookie<'static>>,
     response: http_client::Response,
 }
 
 impl Response {
     /// Create a new instance.
     pub(crate) fn new(response: http_client::Response) -> Self {
-        Self { response }
+        let cookies = response
```

Comment:
Separating the cookies from the rest of the response here means we can hand out `&str`s in the cookie-related methods. Unfortunately, this isn't zero-cost as it will allocate once per received cookie, even if you don't intend on using them at all.